network/address_test.go

```diff
@@ -27,6 +29,12 @@ func TestConnType(t *testing.T) {
 }
 
 func TestAddress(t *testing.T) {
+	localhostNetworkAddresses, _ := net.LookupHost("localhost")
+	sort.Strings(localhostNetworkAddresses)
+	facebookNetworkAddresses, _ := net.LookupHost("facebook.com")
+	sort.Strings(facebookNetworkAddresses)
+	googleNetworkAddresses, _ := net.LookupHost("google.com")
```

Comment:
@nikkolasg - I like the idea of a `Resolver`, however, `LookupHost` is in a package, not in a structure. So would you add another argument to `validHostname` that represents the `Resolver` which would wrap the `LookupHost`-method?

In fact, you could simply pass a `func LookupHost(host string) (addrs []string, err error)` to `validHostname` and either supply `net.LookupHost` (without brackets) or `LookupHostStub` that will be defined in the test.